zanata-model/src/main/java/org/zanata/model/HTextFlowTarget.java

```diff
@@ -446,7 +450,7 @@ public String toString() {
 
     @Transient
     public void clear() {
-        setContents();
+        setContents(null, null, null, null, null, null);
```

Comment:
No the main bug happened because, when we were checking each HTextFlowTarget against removedTargets, equals was returning false (due to other fields being different). This lead to calling `clear()` but I only discovered that via the debugger, because `clear()` didn't work as intended.